api/metrics-api-jaxrs/src/main/java/org/hawkular/metrics/api/jaxrs/handler/MetricHandler.java

```diff
@@ -101,20 +107,34 @@ public void findMetrics(
     @ApiOperation(value = "Add data for multiple metrics in a single call.")
     @ApiResponses(value = {
             @ApiResponse(code = 200, message = "Adding data succeeded."),
+            @ApiResponse(code = 400, message = "Missing or invalid payload.", response = ApiError.class),
```

Comment:
Now I understand what's going on. In `MetricsITest#addMixedDataEmptyRequestPayload`, you set the body to `[]`. But this is not an empty object, this is an empty array. So Jackson fails to deserialize from an array, throws an exception, converted by RestEasy into a 400 reply. 

What you need to do to test the empty object case is `body: new HashMap<>()`.

Also there's the null test case which is missing. And if you add it, you'll get a 500 reply, because of an NPE thrown in this method. You need to check if `metricsRequest` is null and return `emptyPayload` in this case.